### DISCUSSION TITLE
How to filter flat comments?

### DISCUSSION THREAD

**Discussion by kwiat1990** [VirtusLab-Open-Source/strapi-plugin-comments#208] [Q&A] [answered]
I'm facing issues while filtering flat comments and I can't find a solution for it. My code looks like this: 
```ts
filters: {
  $not: [
    {
      threadOf: {
        $null: true,
      },
    },
  ],
},
```

Then for this query Strapi throws following error: `error: select "t0".*, "t0"."id" from "comments_comment" as "t0" where (not (("threadOf" is null)) and (("t0"."removed" is null) or ("t0"."removed" = $1)) and "t0"."related" = $2) order by "t0"."created_at" desc - column "threadOf" does not exist
error: select "t0".*, "t0"."id" from "comments_comment" as "t0" where (not (("threadOf" is null)) and (("t0"."removed" is null) or ("t0"."removed" = $1)) and "t0"."related" = $2) order by "t0"."created_at" desc - column "threadOf" does not exist`.

My initial filter didn't consist `$not` but then Strapi threw another error regarding unallowed operator usage at the root level. Allowed are `$and`, `$or` or `$not`.

**Comment by kwiat1990** [comment]
Solved by [this issue](https://github.com/VirtusLab-Open-Source/strapi-plugin-comments/issues/201). One question though: why for filtering by `threadOf` the filter query must contain `id` field? I don't quite get the reasoning behind it.

On other other hand flat comments contain `related` field, which in my case contains the whole article model, which has a lot of content but I didn't find a way to skip/drop this field while requesting flat comment. Is there any way to do? I thought `related` is a relation field, so it should be only populated if explicitly requested.

**Reply by cyp3rius** [reply, marked as answer]
Yep @kwiat1990, it's a bit tricky as there is already quite complex condition by default there but it is how it is.

In regards to the `related` agreed it might not be necessary to get it with every item. We're going to add possibility to filter it out.

**Reply by kwiat1990** [reply]
Hi @cyp3rius and thanks for the feedback. I’m looking forward to 🙂. If I may ask, what was the reasoning to include it in the comment model in the first place? The endpoint already have all information about the related collection through the URL parameters (collection nam and collection item id), which could be used to fetch, if for some reason needed, a given collection item?

**Comment by enwong** [comment]
> Yep @kwiat1990, it's a bit tricky as there is already quite complex condition by default there but it is how it is.
> 
> In regards to the `related` agreed it might not be necessary to get it with every item. We're going to add possibility to filter it out.

I have encountered the same issue as well, where every comment seems to reiterate the entire information from the article. Is there a method available to filter out `related` content?

**Reply by kwiat1990** [reply]
I didn't find any way at the moment to filter out this info.
